src/main/java/org/datadog/jenkins/plugins/datadog/transport/HttpSender.java

```diff
@@ -0,0 +1,100 @@
+package org.datadog.jenkins.plugins.datadog.transport;
+
+import static org.datadog.jenkins.plugins.datadog.DatadogUtilities.getHttpURLConnection;
+
+import org.datadog.jenkins.plugins.datadog.DatadogUtilities;
+import org.datadog.jenkins.plugins.datadog.util.SuppressFBWarnings;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class HttpSender implements Runnable {
+
+    private static final Logger logger = Logger.getLogger(HttpSender.class.getName());
+
+    private final BlockingQueue<HttpMessage> queue;
+    private final HttpErrorHandler errorHandler;
+    private volatile boolean shutdown;
+
+    HttpSender(final int queueSize, final HttpErrorHandler errorHandler) {
+        this(new LinkedBlockingQueue<HttpMessage>(queueSize), errorHandler);
+    }
+
+    HttpSender(final BlockingQueue<HttpMessage> queue, final HttpErrorHandler errorHandler) {
+        this.queue = queue;
+        this.errorHandler = errorHandler;
+    }
+
+    @SuppressFBWarnings("RV_RETURN_VALUE_IGNORED_BAD_PRACTICE")
+    boolean send(final HttpMessage message){
+        if(!shutdown){
+            queue.offer(message);
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void run() {
```

Comment:
cold you add a comment here explaining how this works?

I'm guessing it polls the queue every second until it's empty or shutdown is triggered but I have no idea. 

Why would this stop if the queue is empty? If we have a very long span that doesn't flush for a while will this be empty and trigger stop?

Reply:
It will stop only if `shutdown=true` and `queue` is empty. I refactored the conditions to be clearer.